packages/common/src/createContract.ts

```diff
@@ -150,13 +158,43 @@ export function createContract<
               }
             >getFunctionParameters(parameters as any);
 
-            const request = {
+            const functionSignature = getAbiItem<Abi, string>({
+              abi,
+              name: functionName,
+            }) as AbiFunction;
+            const functionSelector = getFunctionSelector(functionSignature);
+            if (!functionSignature) throw new Error(`Unable to get function signature for ${functionName}`);
+
+            let request: WriteContractParameters = {
               address,
               abi,
               functionName,
               args,
               ...options,
             };
+            const resourceSelector = await getResourceSelector(functionSelector);
+            const shouldUseCallFrom = resourceSelector && resourceSelector !== staticAbiTypeToDefaultValue.bytes32;
```

Comment:
I see you're doing a falsy check here. Should we update our `getResourceSelector` type to allow for `Promise<Hex | null>`?

Reply:
don't need a falsy check here actually since that's handled in the method and should always return Hex. Will update

Reply:
actually nvm needed falsy check still after making `getResourceSelector ` optional. But don't need to update its return type